src/utilities/theme/legacy-utils.tsx

```diff
@@ -0,0 +1,117 @@
+import tokens from '@shopify/polaris-tokens';
+import {HSLColor} from '../color-types';
+import {colorToHsla, hslToString, hslToRgb} from '../color-transformers';
+import {isLight} from '../color-validation';
+import {constructColorName} from '../color-names';
+import {createLightColor} from '../color-manipulation';
+import {compose} from '../compose';
+
+import {Theme, ColorsToParse, ThemeVariant, ThemeColors, TopBar} from './types';
+import {needsVariantList} from './legacy-config';
+
+export function setColors(theme: Theme | undefined): string[][] | undefined {
+  let colorPairs;
+  if (theme && theme.colors) {
+    Object.entries(theme.colors).forEach(([colorKey, pairs]) => {
+      const colorKeys = Object.keys(pairs as TopBar);
+      if (colorKey === 'topBar' && colorKeys.length > 1) {
+        colorPairs = colorKeys.map((key: string) => {
+          const {topBar} = theme.colors as ThemeColors;
+          const defaultTopBar: TopBar = {background: ''};
+          const colors = topBar ? topBar : defaultTopBar;
```

Comment:
Default topBar added here. In practice, this could would never be reached without a defined `topBar`, but keeping it required has negative impacts on how we write the replacement code.